### PR TITLE
Create an ID3C Glossary

### PR DESCRIPTION
**Pull request by kairstenfay**
Create an ID3C Glossary, initializing it with subpages for Encounters,
Individuals, and Samples.

### CODE REVIEW

**Comment by kairstenfay** [comment]
[Document encounter definitions in wiki](https://trello.com/c/U9jiGOid/710-document-encounter-definitions-in-wiki)

File: Individuals.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+An `individual` in ID3C is "[A single, real person (or other member of a population)](https://github.com/seattleflu/id3c/blob/c5ee5b8d9dbd87a89213f5044a1632cecefd4e7f/schema/deploy/warehouse/individual.sql#L17)."
+A unique Individual identifier is required to create an Individual in ID3C.
+Individuals are required when creating an associated [Encounter] (see [warehouse schema diagram]).
+
+## REDCap projects
+We [created unique, individual identifier strings](https://github.com/seattleflu/id3c-customizations/blob/bdb64db700da9667a8908b425d1a3236e8f5f806/lib/seattleflu/id3c/cli/command/etl/fhir.py#L461) using PII from a participant that we then pass through a [SHA 256 hash function](https://github.com/seattleflu/id3c-customizations/blob/bdb64db700da9667a8908b425d1a3236e8f5f806/lib/seattleflu/id3c/cli/command/clinical.py#L158).
+The identifier string for an individual is created by joining their:
+* [canonicalized name](https://github.com/seattleflu/id3c-customizations/blob/bdb64db700da9667a8908b425d1a3236e8f5f806/lib/seattleflu/id3c/cli/command/etl/fhir.py#L497)*
```
Comment (kairstenfay) [inline Individuals.md:8]: Note to self: remove trailing asterisk here

File: Encounters.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## REDCap projects
+In addition to requiring sufficient information to create an associated [Individual] and Site (i.e. physical place of encounter), we wait to create Encounters in ID3C until the following conditions are met for each project:
+
+### [SCAN]
+- Mail
+  - `consent_form` is complete and verified
+  - `enrollment_questionnaire` is complete and verified
+  - `illness_q_date` is not blank or `back_end_mail_scans` is complete and verified
+  - `post_collection_data_entry_qc` is complete and verified and we can create an associated [Sample], or `post_collection_data_entry_qc` is not complete and verified
```
Comment (joverlee521) [inline Encounters.md:12]: Not required for an encounter to be created. Only required to create a sample.

File: Encounters.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  - `enrollment_date` is not blank
+
+- In-person (kiosk)
+  - `consent_form` is complete and verified
+  - `illness_questionnaire` is complete and verified
+  - `nasal_swab_collection` is complete and verified and we can create an associated Sample, or `nasal_swab_collection` is not complete and verified
```
Comment (joverlee521) [inline Encounters.md:18]: Ditto, not required for an encounter to be created.

File: Encounters.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  - `consent_date` is not blank
+
+### SFS
+- [Kiosk]
+  - `screening` is complete and verified
+  - `main_consent_form` is complete and verified
+  - `enrollment_questionnaire` is complete and verified
+  - `language_questions` != 'Spanish'
+  - we can create an associated [Sample]
```
Comment (joverlee521) [inline Encounters.md:27]: Also requires `enrollment_date` is not blank

File: Encounters.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+- [Swab & Send]
+  - `consent` is complete and verified
+  - `enrollment_questionnaire` is complete and verified
+  - `back_end_mail_scans` is complete and verified
+  - `illness_questionnaire_nasal_swab_collection` is complete and verified
+  - `post_collection_data_entry_qc` is complete and verified
+  - we can create an associated [Sample]
```
Comment (joverlee521) [inline Encounters.md:35]: Requires `enrollment_date_time` to not be blank.

File: Encounters.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  - `enrollment_date` is not blank
+  - we can create an associated [Sample]
+
+- [Swab & Send + Home Flu]
+  - `screening` is complete and verified
+  - `consent` is complete and verified
+  - `back_end_mail_scans` is complete and verified
+  - `illness_questionnaire` is complete and verified
+  - `post_collection_data_entry_qc` is complete and verified
+  - we can create an associated [Sample]
```
Comment (joverlee521) [inline Encounters.md:52]: Requires `enrollment_date` is not blank.

File: Samples.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+A `sample` in ID3C is "[A sample collected from an individual during a specific encounter](https://github.com/seattleflu/id3c/blob/c5ee5b8d9dbd87a89213f5044a1632cecefd4e7f/schema/deploy/warehouse/sample.sql#L23)."
+See the [warehouse schema diagram] to visualize how samples are related to encounters and other concepts in ID3C.
+
+## REDCap projects
+We wait to create Samples in ID3C until the following conditions are met for each project:
+
+### [SCAN]
+- Mail
+  - `post_collection_data_entry_qc` is complete and verified
+  - `return_utm_barcode` is not blank, or `utm_tube_barcode_2` is not blank, or `reenter_barcode` == `reenter_barcode_2`
```
Comment (joverlee521) [inline Samples.md:10]: We do not check `utm_tube_barcode_2` or `reenter_barcode` if `return_utm_barcode` field exists in project.

File: Encounters.md (hunk 2 above)
Comment (kairstenfay) [reply Encounters.md:12]: If `specimen_received` is complete but `specimen_entry` cannot be made (i.e. we can't create an associated Sample), then we "[Skip] enrollment with insufficent [sic] information to construct a specimen".

File: Encounters.md (hunk 3 above)
Comment (kairstenfay) [reply Encounters.md:18]: If `nasal_swab_collection` is complete but `specimen_entry` cannot be made (i.e. we can't create an associated Sample), then we "[Skip] enrollment with insufficent [sic] information to construct a specimen".

File: Samples.md (hunk 7 above)
Comment (kairstenfay) [reply Samples.md:10]: Right, therefore it's an `or` statement where the first condition that is met allows the sample to pass

File: Encounters.md (hunk 2 above)
Comment (joverlee521) [reply Encounters.md:12]: But we also ingest the encounter data without sample if `post_collection_data_entry_qc` is not complete.

File: Samples.md (hunk 7 above)
Comment (joverlee521) [reply Samples.md:10]: I mean if the `return_utm_barcode` field exists at all (blank or not), we do not look at the other fields.

File: Encounters.md (hunk 3 above)
Comment (joverlee521) [reply Encounters.md:18]: We also ingest encounter data without sample if the `nasal_swab_collection` is not complete.

File: Samples.md (hunk 7 above)
Comment (kairstenfay) [reply Samples.md:10]: Ah yes I see that now. Thanks

**Comment by tsibley** [comment]
Thank you again for doing this. :pray: We'll have to try to keep it up-to-date. :crossed_fingers:

**Comment by kairstenfay** [comment]
@tsibley Yes. it won't be easy, but it will be better to maintain this documentation than digging through the code every time we're asked about ID3C vs. REDCap.

**Pull request merged**